util/data_stream.go

```diff
@@ -151,22 +151,11 @@ func NewDataStream(basePath string, p *Package) (*DataStream, error) {
 	if !IsValidRelease(d.Release) {
 		return nil, fmt.Errorf("invalid release: %s", d.Release)
 	}
-	return d, nil
-}
 
-func (d *DataStream) Validate() error {
 	pipelineDir := filepath.Join(d.BasePath, "elasticsearch", DirIngestPipeline)
 	paths, err := filepath.Glob(filepath.Join(pipelineDir, "*"))
 	if err != nil {
-		return err
-	}
-
-	if strings.Contains(d.Dataset, "-") {
-		return fmt.Errorf("data stream name is not allowed to contain `-`: %s", d.Dataset)
-	}
-
-	if !d.validType() {
-		return fmt.Errorf("type is not valid: %s", d.Type)
+		return nil, err
 	}
 
 	if d.Elasticsearch != nil && d.Elasticsearch.IngestPipelineName == "" {
```

Comment:
This is kind of a mix between validation and building the data model. I wonder if we should clean it up later or keep it as is. It would probably mean doing parts of it twice.

Reply:
I'm not sure if I understand your concerns. Any code that modifies a package or a datastream is placed in constructors (e.g. `NewDataStream`). Are you referring to this part of code: https://github.com/elastic/package-registry/blob/b20850de456bf1a0e363163792a7178a4d0c3358/util/data_stream.go#L161-L183 ? If so, then we should come back here and refactor, as the `DataStream.IngestPipeline` is marked as deprecated.

I suppose we can strip this part of at all, unless there is Kibana logic using it.

Reply:
What I initially assume could be an issue does not seem to be one after quickly checking the code. I was worried that for example the pipeline checks do not happen if only `Validate` is run. But this is not the case as before validate is run, the constructor is used which runs all these additional checks always to build the actual package object. So all good.

Reply:
Yeah, I know the design is not really clear as we have to sneak the condition into the `Validate` function, but I didn't see any `Option` to disable validation at `Unpack`.

Reply:
I think I know the person who built this initially and is to blame for the design 😆